spec/models/tip_spec.rb

```diff
@@ -2,51 +2,158 @@
 
 describe Tip do
 
-  let(:user) { FactoryGirl.create(:user) }
-  before { @tip = user.tips.build(title: "Tip", body: "some things", description: "Lorem ipsum") }
+  let(:user) { FactoryGirl.build_stubbed(:user) }
+  let(:tip) { FactoryGirl.build_stubbed(:tip) }
 
-  subject { @tip }
+  subject { tip }
 
   it { should respond_to(:title) }
   it { should respond_to(:description) }
   it { should respond_to(:body) }
   it { should respond_to(:user_id) }
+  it { should respond_to(:slug) }
   it { should respond_to(:user) }
   
-  it { should be_valid }
+  describe 'when all validations are satisfied' do
+    it { should be_valid }
+  end
 
+  describe "associations" do
+    it { should belong_to(:user).counter_cache(true) }
+    it { should have_many(:bookmarks).dependent(:destroy) }
+  end
 
-  describe "when user_id is not present" do
-  	before { @tip.user_id = nil }
-  	it { should_not be_valid }
+  describe "before save" do
+    it "should pygmentize_and_cache_body" do
+      expect(tip).to receive(:pygmentize_and_cache_body)
+      tip.save! 
+    end
   end
 
   describe "with blank title" do
-    before { @tip.title = " "}
+    before { tip.title = " "}
     it { should_not be_valid }
   end
 
   describe "with title that is too long" do
-    before { @tip.title = "a" * 51 }
+    before { tip.title = "a" * 51 }
     it { should_not be_valid }
   end
 
   describe "with blank content" do
-    before { @tip.description = " "}
+    before { tip.description = " "}
     it { should_not be_valid }
   end
 
   describe "with content that is too long" do
-    before { @tip.description = "a" * 251 }
+    before { tip.description = "a" * 251 }
     it { should_not be_valid }
   end
-   describe "with blank content" do
-    before { @tip.body = " "}
+  
+  describe "with blank content" do
+    before { tip.body = " "}
     it { should_not be_valid }
   end
 
-  describe "with content that is too long" do
-    before { @tip.body = "a" * 251 }
-    it { should_not be_valid }
+  describe "#title_and_username" do 
+    context "method creates string that" do
+      it "should be equal" do
```

Comment:
Failure/Error: expect(tip.title_and_username).to eq("#{tip.title} by #{user.username}")

       expected: "Tip title 16 by user_17"
            got: "Tip title 16 by user_16"

       (compared using ==)

Due to the way we've set up the test database, it can't find the correct user when the test is run. To ensure that the right user is found, it's probably best to loop through each one.